packages/fxa-payments-server/src/components/FetchErrorDialogMessage.tsx

```diff
@@ -0,0 +1,23 @@
+import React from 'react';
+import { FetchState } from '../store/types';
+import { APIError } from '../lib/apiClient';
+import DialogMessage from './DialogMessage';
+
+export default ({
+  title,
+  testid = '',
+  fetchState,
+  onDismiss = () => {},
+}: {
+  title: string;
+  testid?: string;
+  fetchState: FetchState<any, APIError>;
+  onDismiss?: Function;
+}) => (
+  <DialogMessage className="dialog-error" onDismiss={onDismiss}>
+    <h4 data-testid={testid}>{title}</h4>
+    {fetchState.error && fetchState.error.message && (
```

Comment:
This new component is basically just `DialogMessage` with added logic to handle FetchState errors